File: omnimer_health_server/src/domain/models/WorkoutFeedback.model.ts
import mongoose, { Schema, Document, Types, Model } from "mongoose";

// Interface cho Workout Feedback
export interface IWorkoutFeedback extends Document {
  _workoutId: Types.ObjectId; // liên kết đến Workout
  suitability?: number; // 1-5
  workout_goal_achieved?: boolean;
  target_muscle_felt?: string;
  injury_or_pain_notes?: string;
  exercise_not_suitable?: boolean;
  additionalNotes?: string;
  createdAt?: Date;
  updatedAt?: Date;
}

const WorkoutFeedbackSchema: Schema<IWorkoutFeedback> = new Schema(
  {
    _workoutId: {
      type: Schema.Types.ObjectId,
      ref: "Workout",
      required: true,
    },
    suitability: { type: Number, min: 1, max: 5 },
    workout_goal_achieved: { type: Boolean },
    target_muscle_felt: { type: String },
    injury_or_pain_notes: { type: String },
    exercise_not_suitable: { type: Boolean },
    additionalNotes: { type: String },
  },
  {
    timestamps: true,
  }
);

export const WorkoutFeedback: Model<IWorkoutFeedback> =
  mongoose.model<IWorkoutFeedback>("WorkoutFeedback", WorkoutFeedbackSchema);

File: omnihealthmobileflutter/lib/data/models/workout/workout_feedback_model.dart
import 'package:omnihealthmobileflutter/domain/entities/workout/workout_feedback_entity.dart';

class WorkoutFeedbackModel extends WorkoutFeedbackEntity {
  WorkoutFeedbackModel({
    super.id,
    required super.workoutId,
    super.suitability,
    super.workoutGoalAchieved,
    super.targetMuscleFelt,
    super.injuryOrPainNotes,
    super.exerciseNotSuitable,
    super.additionalNotes,
    super.createdAt,
  });

  factory WorkoutFeedbackModel.fromJson(Map<String, dynamic> json) {
    return WorkoutFeedbackModel(
      id: json['_id'],
      workoutId: json['workoutId'],
      suitability: json['suitability'],
      workoutGoalAchieved: json['workout_goal_achieved'],
      targetMuscleFelt: json['target_muscle_felt'],
      injuryOrPainNotes: json['injury_or_pain_notes'],
      exerciseNotSuitable: json['exercise_not_suitable'] != null
          ? List<String>.from(json['exercise_not_suitable'])
          : null,
      additionalNotes: json['additionalNotes'],
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'workoutId': workoutId,
      'suitability': suitability,
      'workout_goal_achieved': workoutGoalAchieved,
      'target_muscle_felt': targetMuscleFelt,
      'injury_or_pain_notes': injuryOrPainNotes,
      'exercise_not_suitable': exerciseNotSuitable,
      'additionalNotes': additionalNotes,
    };
  }
}
